### PR TITLE
ISPN-7906 Infinispan Query DSL does not handle inheritance of propert…

### PR DESCRIPTION
**Pull request by anistor**
…ies/fields correctly

https://issues.jboss.org/browse/ISPN-7906

### CODE REVIEW

**Comment by anistor** [comment]
@lucasponce The property lookup issue is fixed now and will be released pretty soon.

File: object-filter/src/main/java/org/infinispan/objectfilter/impl/util/ReflectionHelper.java [hunk 1]
```diff
@@ -250,13 +252,26 @@ public static PropertyAccessor getAccessor(Class<?> clazz, String propertyName)
...
       }
-      if (propertyName.contains(".")) {
+      if (propertyName.indexOf('.') != -1) {
          throw new IllegalArgumentException("The argument cannot be a nested property name");
       }
+      String propertyNameSuffix = Character.toUpperCase(propertyName.charAt(0)) + propertyName.substring(1);
+
+      Class<?> c = clazz;
+      while (c != null) {
+         PropertyAccessor m = getAccessor(c, propertyName, propertyNameSuffix);
```
Comment (slaskawi) [inline object-filter/src/main/java/org/infinispan/objectfilter/impl/util/ReflectionHelper.java]: OMG :fearful:

**Comment by slaskawi** [comment]
Let's wait for CI results :)

File: object-filter/src/main/java/org/infinispan/objectfilter/impl/util/ReflectionHelper.java (hunk 1 above)
Comment (anistor) [inline object-filter/src/main/java/org/infinispan/objectfilter/impl/util/ReflectionHelper.java]: omg what?

Comment (slaskawi) [inline object-filter/src/main/java/org/infinispan/objectfilter/impl/util/ReflectionHelper.java]: hahah no no, it's ok.. I know there is no better way to do it. But such constructs are always scary :D

**Comment by galderz** [comment]
Last build did not run, trying again

**Comment by gustavonalle** [comment]
Running the tests locally...

**Comment by gustavonalle** [comment]
Integrated on master, thanks @anistor

**Comment by anistor** [comment]
@gustavonalle You closed it but I I do not see the commits in master. Did you forget to push? I'll merge this myself.

**Comment by gustavonalle** [comment]
Sorry @anistor

**Pull request closed without merging**